alacritty_terminal/src/tty/windows/winpty.rs

```diff
@@ -137,24 +116,26 @@ pub fn new<C>(config: &Config<C>, size: &SizeInfo, _window_id: Option<usize>) ->
     winpty.spawn(&spawnconfig).unwrap();
 
     let child_watcher = ChildExitWatcher::new(winpty.raw_handle()).unwrap();
-    let agent = Agent::new(winpty);
-
-    Pty {
-        handle: super::PtyHandle::Winpty(WinptyHandle::new(agent)),
-        conout: super::EventedReadablePipe::Named(conout_pipe),
-        conin: super::EventedWritablePipe::Named(conin_pipe),
-        read_token: 0.into(),
-        write_token: 0.into(),
-        child_event_token: 0.into(),
-        child_watcher,
-    }
+    let agent = WinptyAgent {
+        winpty: Arc::new(winpty),
+        conout: EventedReadablePipe::Named(conout_pipe),
+        conin: EventedWritablePipe::Named(conin_pipe),
+    };
+
+    Pty::new(super::PtyHandle::Winpty(agent), child_watcher)
 }
 
-impl OnResize for Winpty {
+impl OnResize for WinptyResizeHandle {
     fn on_resize(&mut self, sizeinfo: &SizeInfo) {
         let (cols, lines) = (sizeinfo.cols().0, sizeinfo.lines().0);
         if cols > 0 && cols <= u16::MAX as usize && lines > 0 && lines <= u16::MAX as usize {
-            self.set_size(cols as u16, lines as u16)
+            // This unsafe cast from const to non-const is actually thread-safe because
```

Comment:
```suggestion
            // This unsafe cast from const to non-const is thread-safe because
```

Just a super, super minor thing, but I think the 'actually' is a bit strange here. We're just stating facts here, we're not trying to prove anyone wrong.